api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -151,7 +158,45 @@ public DbUser updateUserWithRetries(Function<DbUser, DbUser> userModifier, DbUse
     }
   }
 
-  private void updateDataAccessLevel(DbUser user) {
+  private void updateDataAccessLevel(DbUser dbUser) {
+    final DataAccessLevel previousDataAccessLevel = dbUser.getDataAccessLevelEnum();
```

Comment:
Could you please split the audit changes and this into separate PRs? There's a lot of content already in the PR without these changes and I don't see how this relates to the rest.

Reply:
This refactoring gives me a single place to call `fireUpdateDataAccessActon()`, as now there's only one place we call `setDataAccessLevelEnum()`.

Additionally, I found the existing function a bit difficult to follow and tried to squeeze out some duplication so I could have smaller methods and understand what I was working with.

Reply:
I see the connection, I just think it could be split up into multiple PRs, e.g. UserService and the corresponding tests. If I'm missing something and they are too deeply intertwined - fine. I will take a look through the rest of this PR later today given the size.

Reply:
I looked over the changes to `updateDataAccessLevel` and I think they're correct. I like the changes and I think its more readable than what we had before but to echo @calbach 's sentiments, it can be hard to give a good review when there's a lot to cover.

Splitting up the necessary refactorings into its own PR can make it easier on us to give a thorough review, especially for critical code paths like this one.

Reply:
Yeah, this one got away from me a bit, and it wasn't helped when I merged RW-4009 into this branch. Lots of folks were out so it sat for a while. Plus I was on the home stretch and ready to have all this behind me. I'll try to keep them shorter in the future.

I'll try to split this out, but if I can't get it in 30 minutes I don't know if it's worth it.

Reply:
PR is up at #2911

I haven't reverted any of the changes on this branch yet though; waiting on that to merge first.